walk/cache/cache.go

```diff
@@ -0,0 +1,78 @@
+package cache
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io/fs"
+	"os"
+	"time"
+
+	"github.com/adrg/xdg"
+	bolt "go.etcd.io/bbolt"
+)
+
+type Entry struct {
+	Size     int64
+	Modified time.Time
+}
+
+func (e *Entry) HasChanged(info fs.FileInfo) bool {
+	return !(e.Modified == info.ModTime() && e.Size == info.Size())
+}
+
+func Open(root string, temp bool) (*bolt.DB, error) {
+	var err error
+	var path string
+
+	if temp {
+		// If no cache is desired, rather than complicate the logic for cache / no cache, we just create a cache instance
+		// backed by a temporary file instead
+		if f, err := os.CreateTemp("", "treefmt-no-cache-*.db"); err != nil {
+			return nil, fmt.Errorf("failed to create a temporary db file: %w", err)
+		} else {
+			path = f.Name()
+		}
+	} else {
+		// Otherwise, the database will be located in `XDG_CACHE_DIR/treefmt/eval-cache/<name>.db`, where <name> is
+		// determined by hashing the treeRoot path.
+		// This associates a given treeRoot with a given instance of the cache.
+		h := sha1.New()
+		h.Write([]byte(root))
+		digest := h.Sum(nil)
```

Comment:
Since Go 1.2 you can now do:
```suggestion
		digest := sha1.Sum([]byte(root))
```